mephisto/abstractions/providers/mturk/mturk_utils.py

```diff
@@ -738,3 +743,111 @@ def expire_and_dispose_hits(
             h["dispose_exception"] = e
             non_disposed_hits.append(h)
     return non_disposed_hits
+
+
+def try_prerun_cleanup(db: "MephistoDB", requester_name: str) -> None:
+    """
+    Try to see if there are any outstanding HITS for the given requester, and LOUDLY WARN if
+    there are any, allowing the user to run a cleanup in-line.
+    """
+    cleanups_path = os.path.join(
+        DEFAULT_CONFIG_FOLDER, "mturk_requesters_last_cleanups.json"
+    )
+    last_cleanup_times = {}
+    if os.path.exists(cleanups_path):
+        with open(cleanups_path) as cleanups_file:
+            last_cleanup_times = json.load(cleanups_file)
+
+    last_cleanup = last_cleanup_times.get(requester_name, 0)
+
+    if time.time() - last_cleanup < CLEANUP_CHECK_TIME:
+        # Too recent, no cleanup needed
+        return
+
+    print(
+        f"It's been more than a day since you last ran a job "
+        f"with {requester_name}, checking for outstanding tasks..."
+    )
+
+    requester = db.find_requesters(requester_name=requester_name)[0]
+    client = requester._get_client(requester._requester_name)
+
+    def hit_is_broken(hit: Dict[str, Any]) -> bool:
+        return (
+            hit["NumberOfAssignmentsCompleted"] == 0
+            and hit["HITStatus"] != "Reviewable"
+        )
+
+    query_time = time.time()
+    outstanding_hit_types = get_outstanding_hits(client)
+    if time.time() - query_time > 60:
+        print(
+            "That took a while! You may want to run `mephisto scripts "
+            "cleanup mturk` later to clear out some of the older HIT types.\n"
+        )
+    broken_hit_types = {
+        k: [h for h in v if hit_is_broken(h)]
+        for (k, v) in outstanding_hit_types.items()
+        if any([h for h in v if hit_is_broken(h)])
+    }
+    num_hit_types = len(broken_hit_types.keys())
```

Comment:
Is it helpful to printout these types with their counts? Maybe some insight on something being wrong?

Reply:
NM, I see it is down below.